packages/vite/src/node/optimizer/registerMissing.ts

```diff
@@ -54,10 +54,12 @@ export function createMissingImporterRegisterFn(
       knownOptimized = newData!.optimized
 
       // update ssr externals
-      server._ssrExternals = resolveSSRExternal(
-        server.config,
-        Object.keys(knownOptimized)
-      )
+      if (server.config.ssr) {
```

Comment:
The `ssr` param in `rerun` is what indicates if the server is running in ssr mode. @bluwy could you confirm this, I see that you approved this one
```suggestion
      if (ssr) {
```

Reply:
Actually, I'm not sure if we can trust this param. As far as I get this, `tryNodeResolve` is the only invoker of `registerMissing`. An important detail I found about `tryNodeResolve` is that this function is not always called with `ssr === true` even when SSR is enabled. See how `ssrModuleLoader` calls `tryNodeResolve`, for example:

https://github.com/vitejs/vite/blob/e6495f0a52c9bd2cae166934dc965f8955ce035d/packages/vite/src/node/ssr/ssrModuleLoader.ts#L222

The problem is, I don't know 100% how it is intended to work since there are no tests. I just tried to be as close to the original implementation as possible.

Maybe it will be indeed sufficient to call `resolveSSRExternal` only for the case `ssr === true`. Let me know if you confirm it.

Reply:
@aleclarson could you help us checking this one?

Reply:
My bad. I think you're right @patak-dev, that slipped off my mind. @vursen regarding your concern, that seems like it should be a `true` instead, I'm not sure why it's false since it's used by hookNodeResolve. But yes, we do lack tests for this case, and @aleclarson may know more about it.

Reply:
The `ssr` argument is actually undefined in the call above, not false. The false value is for the `targetWeb` argument. :)

I think we can use `ssr` instead of `server.config.ssr` (as suggested by @patak-dev), since the `tryNodeResolve` call will be removed in #6591.

Reply:
OK, updated this line. Please, take a look.